multi: convert the existing channeldb.ChannelType uint8 into a bit field

Roasbeef opened the pull request:
In this commit, we convert the existing `channeldb.ChannelType` type
into a _bit field_. This doesn't require us to change the current
serialization or interpretation or the type as it is, since all the
current defined values us a distinct bit. This PR lays the ground work
for any future changes that may introduce new channel types (like anchor
outputs), and also any changes that may modify the existing invariants
around channels (if we're the initiator, we always have the funding
transaction).

cfromknecht commented on channeldb/channel.go:
```diff
@@ -125,39 +125,46 @@ var (
-// TODO(roasbeef): split up per-chain?
+// fee negotiation, channel closing, the format of HTLCs, etc. Structure-wise,
+// a ChannelType is a bit field, with each bit denoting a modification from the
+// base channel type of single funder.
```
this last bit isn't quite accurate since dual and single funder each have their own bit

cfromknecht commented on channeldb/channel.go:
```diff
@@ -125,39 +125,46 @@ var (
-	// SingleFunder represents a channel wherein one party solely funds the
-	// entire capacity of the channel.
-	SingleFunder ChannelType = 0
+	// SingleFunderBit represents a channel wherein one party solely funds
```
would personally refer to these as `Flags` rather than `Bits`, but not a blocker

cfromknecht commented on channeldb/channel.go:
```diff
@@ -125,39 +125,46 @@ var (
+}
+
+// IsDualFunder returns true if the ChannelType has the DualFunderBit set.
+func (c ChannelType) IsDualFunder() bool {
```
order helper methods in the order the flags are defined?

guggero commented:
Travis fails for the dual funder flow:

```
--- FAIL: TestLightningWallet (356.83s)
    --- PASS: TestLightningWallet/btcwallet/bitcoind:change_output_spend_confirmation (6.29s)
    --- PASS: TestLightningWallet/btcwallet/bitcoind:insane_fee_reject (0.01s)
    --- PASS: TestLightningWallet/btcwallet/bitcoind:single_funding_workflow (0.72s)
    --- PASS: TestLightningWallet/btcwallet/bitcoind:single_funding_workflow_tweakless (2.46s)
    --- FAIL: TestLightningWallet/btcwallet/bitcoind:dual_funder_workflow (0.33s)
        interface_test.go:489: unable to consume alice's sigs: counterparty's commitment signature is invalid
```

cfromknecht commented on channeldb/channel.go:
```diff
@@ -125,39 +125,46 @@ var (
-	SingleFunder ChannelType = 0
+	// SingleFunderBit represents a channel wherein one party solely funds
+	// the entire capacity of the channel.
+	SingleFunderBit ChannelType = 0
```
heh, now there is no single funder bit

Roasbeef replied:
Yeah the old one was incorrect, in that it started things off at zero instead of 1, so it wasn't backwards compatible.

Roasbeef commented:
Fixed the test failure. An earlier version was incorrect in that it re-defined the base channel type (single funder) to one instead of zero. I fixed that, but then introduced another bug that would cause it to not recognize dual funder channels properly. This has been fixed now, by defining `IsSingleFunder` using the dual funder bit.
